core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -71,7 +71,7 @@ class KafkaApis(val requestChannel: RequestChannel,
   def handle(request: RequestChannel.Request) {
     try {
       trace("Handling request:%s from connection %s;securityProtocol:%s,principal:%s".
-        format(request.requestObj, request.connectionId, request.securityProtocol, request.session.principal))
+        format(Option(request.requestObj).getOrElse(request.header), request.connectionId, request.securityProtocol, request.session.principal))
```

Comment:
It may be worth doing it the other way around, as most requests are the new style now. Well spotted.